components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -35,17 +39,18 @@ class ObserverRegistry<T> : Observable<T> {
     }
 
     override fun register(observer: T, owner: LifecycleOwner, autoPause: Boolean) {
-        if (owner.lifecycle.currentState == Lifecycle.State.DESTROYED) {
+        // Don't register if the owner is already destroyed
+        if (!owner.lifecycle.currentState.isAtLeast(CREATED)) {
             return
```

Comment:
Is `currentState` something we can't access anymore? This is still what `LiveData` does internally:
https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/lifecycle/livedata-core/src/main/java/androidx/lifecycle/LiveData.java#190

Reply:
We can still access it, but `isAtLeast` covers the `INITIALIZED` state too. If `LiveData` just uses equals internally I could switch back.

Reply:
I wonder if there can be a situation where we pass in a `LifecycleOwner` that is not CREATED yet and then we would never register? 🤔

Reply:
[During onCreate](https://stackoverflow.com/questions/45459036/lifecycle-state-always-initialized), before the `ON_CREATE` event is triggered. However, it's probably more sensible to allow the observer to be registered during that time.